ui/pages/confirmations/components/confirm/info/contexts/advanced-details-context.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {
+  ReactElement,
+  createContext,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { useSelector } from 'react-redux';
+import { getSendHexDataFeatureFlagState } from '../../../../../../ducks/metamask/metamask';
+import { getUseNonceField } from '../../../../../../selectors';
+
+type AdvancedDetailsHandlerContextType = {
+  showAdvancedDetails: boolean;
+  setShowAdvancedDetails: (value: boolean) => void;
+};
+
+export const AdvancedDetailsHandlerContext = createContext<
+  AdvancedDetailsHandlerContextType | undefined
+>(undefined);
+
+export const AdvancedDetailsProvider: React.FC<{
+  children: ReactElement;
+}> = ({ children }) => {
+  const enableCustomNonce = useSelector(getUseNonceField);
+  const showHexData = useSelector(getSendHexDataFeatureFlagState);
+
+  const [showAdvancedDetails, setShowAdvancedDetails] = useState(
```

Comment:
I understood the final decision on this was to add a new property in the `PreferenceController` state to persist the advanced toggle preference.

And we'd do a one-off migration for anyone with custom name or hex data visible at the time, to set this new preference.